src/launchpad/parsers/android/dex/dex_class_parser.py

```diff
@@ -98,19 +114,149 @@ def _get_annotations_directory(self) -> AnnotationsDirectory | None:
+            cursor = self._buffer_wrapper.cursor
+            field_idx_diff = self._buffer_wrapper.read_uleb128()
+            access_flags = self._buffer_wrapper.read_uleb128()
+            field_overhead = self._buffer_wrapper.cursor - cursor
+
+            field_index = last_index + field_idx_diff
+            last_index = field_index
+
+            if field_index < len(self._static_field_values):
+                initial_value = self._static_field_values[field_index]
+            else:
+                initial_value = None
+
+            field_parser = DexFieldParser(
+                self._buffer_wrapper,
+                self._header,
+                field_index,
+                initial_value,
+                field_overhead,
+                self._get_annotations_directory(),
+            )
+
+            field = Field(
+                size=field_parser.get_size(),
+                signature=field_parser.get_signature(),
+                access_flags=DexBaseUtils.parse_access_flags(access_flags),
+                annotations=field_parser.get_annotations(),
+            )
+
+            fields.append(field)
+
+        self._buffer_wrapper.seek(cursor)
+        self._static_fields = fields
+        return fields
+
+    def _get_instance_fields(self) -> list[Field]:
+        # Cache for later reuse
+        if self._instance_fields is not None:
+            return self._instance_fields
+
+        if self._class_data_offset == 0:
+            return []
+
+        cursor = self._buffer_wrapper.cursor
+        self._buffer_wrapper.seek(self._class_data_offset)
+
+        static_fields_size = self._buffer_wrapper.read_uleb128()
+        instance_fields_size = self._buffer_wrapper.read_uleb128()
+        self._buffer_wrapper.read_uleb128()  # direct_methods_size
+        self._buffer_wrapper.read_uleb128()  # virtual_methods_size
+
+        # Skip static fields
+        for _ in range(static_fields_size):
+            self._buffer_wrapper.read_uleb128()
+            self._buffer_wrapper.read_uleb128()
+
+        # Parse instance fields
+        fields = []
+        last_index = 0
+
+        for _ in range(instance_fields_size):
+            field_cursor = self._buffer_wrapper.cursor
+            field_idx_diff = self._buffer_wrapper.read_uleb128()
+            access_flags = self._buffer_wrapper.read_uleb128()
+            field_overhead = self._buffer_wrapper.cursor - field_cursor
+
+            field_index = last_index + field_idx_diff
+            last_index = field_index
+
+            field_parser = DexFieldParser(
+                buffer_wrapper=self._buffer_wrapper,
+                header=self._header,
+                field_index=field_index,
+                initial_value=None,  # Instance fields will always have a null initial value
+                field_overhead=field_overhead,
+                annotations_directory=self._get_annotations_directory(),
+            )
+
+            field = Field(
```

Comment:
nit: DexFieldParser might as well have a `field = field_parser.parse()` method

Reply:
Semi-disagree here as the caller needs some info about the offsets and positions around it, so would take quite a few args to pass to parse